base/ca/src/main/java/com/netscape/cms/authentication/CMCAuthForEST.java

```diff
@@ -0,0 +1,117 @@
+
+import com.netscape.certsrv.authentication.AuthCredentials;
+import com.netscape.certsrv.base.EBaseException;
+import com.netscape.certsrv.base.SessionContext;
+
+/**
+ * EST CMC Authentication.
+ *
+ * Extends CMCAuth to support EST-forwarded CMC requests where the actual
+ * EST client certificate is passed via HTTP header pki-est-client-cert
+ * instead of via TLS (SessionContext.SSL_CLIENT_CERT).
+ *
+ * The TLS connection between EST and CA uses the EST subsystem certificate,
+ * but the CMC request is signed by the EST client's certificate. This
+ * authenticator verifies that the CMC signer matches the EST client cert
+ * from the header.
+ *
+ * @author cfu
+ */
+public class CMCAuthForEST extends CMCAuth {
+
+    public static org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(CMCAuthForEST.class);
+
+    public static final String EST_CLIENT_CERT_HEADER = "pki-est-client-cert";
+
+    /**
+     * Gets the client certificate for CMC authentication.
+     *
+     * For EST-forwarded requests, this reads the EST client certificate from
+     * the pki-est-client-cert HTTP header instead of from the TLS session context.
+     *
+     * @param auditContext The session context
+     * @return The EST client certificate, or null if not available
+     * @throws RuntimeException if the certificate header exists but cannot be decoded/parsed
+     */
+    protected X509Certificate getClientCertificate(SessionContext auditContext) {
+        String method = "CMCAuthForEST.getClientCertificate: ";
+
+        // First try to get EST client cert from HTTP header
+        Object estClientCertObj = auditContext.get(EST_CLIENT_CERT_HEADER);
+
+        if (estClientCertObj != null && estClientCertObj instanceof String) {
+            String estClientCertB64 = (String) estClientCertObj;
+            logger.debug(method + "Found pki-est-client-cert header");
+
+            try {
+                // Decode base64 certificate
+                byte[] certBytes = Utils.base64decode(estClientCertB64);
+
+                // Parse certificate as JSS X509CertImpl (required for CMCAuth)
+                X509CertImpl cert = new X509CertImpl(certBytes);
+
+                logger.debug(method + "Successfully decoded EST client certificate: " +
+                    cert.getSubjectDN().getName());
+                return cert;
+
+            } catch (Exception e) {
+                // Certificate exists but cannot be parsed - this is a real error
+                logger.error(method + "Failed to decode EST client certificate from header: " +
+                    e.getMessage(), e);
+                throw new RuntimeException("Failed to decode EST client certificate", e);
+            }
+        }
+
+        // No EST client certificate header found
+        logger.debug(method + "No pki-est-client-cert header found");
+        return null;
+    }
+
+    @Override
+    public AuthToken authenticate(AuthCredentials credentials)
```

Comment:
Using this profile the EST agent certificate/credentials are not verified. Only the CMC certificate are authorised. Is this the goal or this method should verify first that the request is coming from an authorised agent and then verify the credentials used for the CMC request?